medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -63,15 +64,33 @@ impl Error for RetrieveFromQueueError {
     }
 }
 
+pub struct RetrievedMessage<T> {
+    pub message: T,
+    peek_lock: PeekLockResponse,
+}
+
+impl<T> RetrievedMessage<T> {
+    pub async fn remove(&self) -> Result<String, anyhow::Error> {
+        let queue_removal_result = self.peek_lock.delete_message().await.map_err(|e| {
+            tracing::error!("{:?}", e);
+            anyhow!("Queue Removal Error")
+        });
+        tracing::info!("Removed job from ServiceBus ({:?})", queue_removal_result);
+        queue_removal_result
+    }
+}
+
 impl DocIndexUpdaterQueue {
     fn new(service_bus: Client) -> Self {
         Self { service_bus }
     }
 
-    pub async fn receive<T: Message>(&mut self) -> Result<T, RetrieveFromQueueError> {
-        let message = self
+    pub async fn receive<T: Message>(
+        &mut self,
+    ) -> Result<RetrievedMessage<T>, RetrieveFromQueueError> {
+        let peek_lock = self
             .service_bus
-            .peek_lock(time::Duration::days(1), Some(time::Duration::seconds(10)))
+            .peek_lock_full(time::Duration::days(1), Some(time::Duration::seconds(10)))
```

Comment:
What's `peek_lock_full`? I can't find anything about it in the Rust SDK.

Reply:
From the PR description:

> Relates to [AzureSDKForRust#247](https://github.com/MindFlavor/AzureSDKForRust/pull/247/files).

It's only in our branch (which I had to adjust for this ticket) that this concept of a `peek_lock` variant which brings back a full `PeekLockResponse` (which I also had to create) exists.